unicorn/include/cocaine/idl/unicorn.hpp

```diff
@@ -0,0 +1,267 @@
+        */
+        typedef boost::mpl::list<
+            path_t
+        > argument_type;
+
+        /**
+        * current version in ZK
+        */
+        typedef stream_of<
+            versioned_value_t
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct del {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "del";
+        }
+
+        /**
+        * delete node. Will only succeed if there are no child nodes.
+        */
+        typedef boost::mpl::list<
+            path_t,
+            version_t
+        > argument_type;
+
+        typedef option_of<
+            bool
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct increment {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "increment";
+        }
+
+        /**
+        * increment node value in path by passed value.
+        * If either passed or stored value is not numeric will return error.
+        * If one of the values is float - result value will be float.
+        */
+        typedef boost::mpl::list<
+            path_t,
+            value_t
+        > argument_type;
+
+        /**
+        * return value after increment
+        */
+        typedef option_of <
+            versioned_value_t
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct lsubscribe {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "lsubscribe";
+        }
+
+        typedef boost::mpl::list<
+            path_t
+        > argument_type;
+
+        typedef stream_of<
+            version_t,
+            std::vector<std::string>
+        >::tag upstream_type;
+
+        typedef unicorn_final_tag dispatch_type;
+    };
+
+    struct lock {
+        typedef unicorn_tag tag;
+
+        static const char* alias() {
+            return "lock";
+        }
+
+        typedef unicorn_locked_tag dispatch_type;
+
+        typedef boost::mpl::list<
+            path_t
+        > argument_type;
+
+        typedef option_of <
+            bool
+        >::tag upstream_type;
+    };
+
+    struct unlock {
+        typedef unicorn_locked_tag tag;
+        static const char* alias() {
+            return "unlock";
+        }
+    };
+
+    struct close {
+        typedef unicorn_final_tag tag;
+        static const char* alias() {
+            return "close";
+        }
+    };
+};
+
+template<>
+struct protocol<unicorn_tag> {
+    typedef boost::mpl::int_<
+        1
+    >::type version;
+
+    typedef boost::mpl::list<
+        unicorn::subscribe,
+        unicorn::lsubscribe,
+        unicorn::put,
+        unicorn::create,
+        unicorn::del,
+        unicorn::increment,
+        unicorn::lock
+    > messages;
+
+    typedef unicorn type;
```

Comment:
Replace `type -> scope` here and elsewhere. For example https://github.com/cocaine/cocaine-core/blob/master/include/cocaine/idl/storage.hpp#L128